sdk/resourcemanager/azure-resourcemanager-redis/src/main/java/com/azure/resourcemanager/redis/implementation/RedisFirewallRulesImpl.java

```diff
@@ -41,14 +44,26 @@
     }
 
     public void addRule(RedisFirewallRuleImpl rule) {
```

Comment:
Is these method guaranteed to be called only by sync method (at least from us)?

Reply:
We didn't use it in async method. But it can't be guaranteed when using by customer.

After discussion, the `ExternalChildResourcesCachedImpl` implementation would be removed later, due to the cache is blocking.

Reply:
Issue add https://github.com/Azure/azure-libraries-for-java/issues/1261